source/link/linker.cpp

```diff
@@ -157,6 +157,10 @@ spv_result_t Linker::Link(const std::vector<std::vector<uint32_t>>& binaries,
     module->ForEachInst([&id_bound](Instruction* insn){
       for (auto& o : *insn)
         if (spvIsIdType(o.type)) o.words[0] += id_bound;
+      uint32_t result_id = insn->result_id();
+      if (result_id != 0u) insn->SetResultId(result_id + id_bound);
```

Comment:
Optional:  A little coding trick.  You can push the definition of result_id into the conditional and make the check against 0 implicit.   

```
if (auto result_id = insn->result_id()) 
   insn->SetResultId(result_id + id_bound);
```

The chief advantages are: one less line of code to read, and the result_id variable goes out of scope right away, so you can't misuse it too much.

If the lambda's body were much longer, I'd ask for result_id and result_type to be made const, because they can.